search-service/src/main/java/de/cxp/ocs/elasticsearch/facets/CategoryFacetCreator.java

```diff
@@ -79,13 +83,21 @@ protected boolean isMatchingFilterType(InternalResultFilter internalResultFilter
 
 		Map<String, HierarchialFacetEntry> entries = new LinkedHashMap<>(catBuckets.size());
 		long absDocCount = 0;
+		boolean isFiltered = isMatchingFilterType(intFacetFilter);
 
 		for (Bucket categoryBucket : catBuckets) {
 			String categoryPath = categoryBucket.getKeyAsString();
 
 			String[] categories = StringUtils.split(categoryPath, '/');
-			// TODO: in case a category is filtered, it might be a good idea to
-			// only show the according path
+			
+			if(isFiltered) {
+				Map<String, InternalResultFilter> filters = linkBuilder.getFilters();
+				String[] categoryValues = filters.get(CATEGORY_KEY).getValues();
```

Comment:
I think you can use `intFacetFilter` here and don't have to fetch it from all filters again. Then you also don't need the CATEGORY_KEY - because this is just specific for your usecase. It's also possible to index other hierarchical data, not just "category".

Reply:
Yes, you are right, I've changed it and use now intFacetFilter.